### DISCUSSION TITLE
Is it possible to run an entire SQL file with this library?

### DISCUSSION THREAD

**Discussion by lmeerwood** [oracle/python-oracledb#17] [Q&A] [answered]
Hey all,

My company is moving to Airflow to handle our ETL pipeline. Airflow is about to start using the new OracleDB libary, moving away from CX-Oracle. 

One thing I was wondering is: Is it possible to run an entire Oracle SQL file with this library? In a lot of our SQL files we have multiple statements mixed with Stored Procedures. Ideally I'd love to just pass it in and have Airflow/OracleDB parse and run the file. 

My current solution is to split up the file into individual statements and run them as either an SQL statement or a Stored Procedure, but this is a bit time consuming and feels "wrong".

**Comment by cjbj** [comment, marked as answer]
Yes, it's possible but it needs to be done in Python. It's not part of the driver itself.

See how the example SQL schema is installed by reading from a simplified SQL file in [sample_env.py](https://github.com/oracle/python-oracledb/blob/v1.0.1/samples/sample_env.py#L171-L204)

Also see the sample [sqlp.py](https://github.com/oracle/python-oracledb/blob/v1.0.1/samples/sqlp.py) which reads interactively, is more tolerant of SQL / SQL*Plus / PL/SQL syntax but is missing some useful things like substitution and bind variable support.

Pull Requests for improvements are welcome.

**Reply by lmeerwood** [reply]
Awesome. thanks for the feedback. I'll have a look at those and see what I can do!
